invideoquiz/public/js/src/invideoquiz.js

```diff
@@ -0,0 +1,109 @@
+/* Javascript for InVideoQuizXBlock. */
+function InVideoQuizXBlock(runtime, element) {
+    var videoId = $('.in-video-quiz-block').data('videoid');
+    var studentMode = $('.in-video-quiz-block').data('mode') !== 'staff';
+    var problemTimesMap = InVideoQuizXBlock.config[videoId] || {};
+    var extraVideoButton = '<button class="in-video-continue">Continue</button>';
+    var video;
+    var videoState;
+    var intervalTime = 500; // interval at which to check for problems to display
+    var intervalTimeout = 1500; // timeout to wait before checking for problems again after "play" is clicked
+
+    // Main wrapper function
+    $(function () {
+        $('.in-video-quiz-block').closest('.vert').hide();
+
+        if (!videoId || !problemTimesMap) {
+            return;
+        }
+
+        $('#seq_content .vert-mod .vert').each(function () {
+            var component = $(this);
+
+            if (studentMode) {
+                setUpStudentView(component);
+            } else {
+                showProblemTimesToInstructor(component);
+            }
+        });
+
+        if (studentMode) {
+          bindVideoEvents();
+        }
+    });
+
+    // Set Up In Video problems for students
+    function setUpStudentView(component) {
+        var componentIsVideo = component.data('id').indexOf(videoId) !== -1;
+        if (componentIsVideo) {
+            video = $('.video', component);
+            videoState = video.data('video-player-state');
+        } else {
+            $.each(problemTimesMap, function (time, componentId) {
+                if (component.data('id').indexOf(componentId) !== -1) {
+                    component.addClass('in-video-problem-wrapper');
+                    $('.xblock-student_view', component).append(extraVideoButton).addClass('in-video-problem').hide();
+                }
+            });
+        }
+    }
+
+    // Mark in video problems and display at what time they display in the video
+    function showProblemTimesToInstructor(component) {
+        $.each(problemTimesMap, function (time, componentId) {
+            var isInVideoComponent = component.data('id').indexOf(componentId) !== -1;
+            if (isInVideoComponent) {
+                var minutes = parseInt(time / 60, 10);
+                var seconds = ('0' + (time % 60)).slice(-2);
+                var timeParagraph = '<p class="in-video-alert"><i class="fa fa-exclamation-circle"></i>\
```

Comment:
http://stackoverflow.com/questions/805107/creating-multiline-strings-in-javascript